# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: docshell/base/CanonicalBrowsingContext.cpp
# Commit: 787c24b748c8
# Full Hash: 787c24b748c83b624c42ab1db9d22c9d9b086153
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-04-10 03:36:58
# Description:
#   Bug 1702714 - GetBrowserBridgeParent may be null after destroy, r=farre
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D110827
# ==============================================================================

diff -r 6f298a911d9c -r 787c24b748c8 docshell/base/CanonicalBrowsingContext.cpp
--- a/docshell/base/CanonicalBrowsingContext.cpp	Fri Apr 09 17:06:20 2021 +0000
+++ b/docshell/base/CanonicalBrowsingContext.cpp	Fri Apr 09 17:10:18 2021 +0000
@@ -1271,7 +1271,8 @@
   bool wasRemote = oldBrowser && oldBrowser->GetBrowsingContext() == target;
   if (wasRemote) {
     MOZ_DIAGNOSTIC_ASSERT(oldBrowser != embedderBrowser);
-    MOZ_DIAGNOSTIC_ASSERT(oldBrowser->GetBrowserBridgeParent());
+    MOZ_DIAGNOSTIC_ASSERT(oldBrowser->IsDestroyed() ||
+                          oldBrowser->GetBrowserBridgeParent());
 
     // `oldBrowser` will clear the `UnloadingHost` status once the actor has
     // been destroyed.
